common/djangoapps/adaptive_learning/config/models.py

```diff
@@ -3,15 +3,72 @@
 """
 
 from config_models.models import ConfigurationModel
+from openedx.core.djangoapps.xmodule_django.models import CourseKeyField
 
 
 class AdaptiveLearningEnabledFlag(ConfigurationModel):
     """
-    A flag that enables/disables Adaptive Learning across an entire instance.
+    A flag that enables/disables the Adaptive Learning feature across an entire instance.
+
+    Defaults to False.
     """
 
+    @classmethod
+    def feature_enabled(cls, course_id=None):
+        """
+        Looks at the currently active configuration model to determine
+        whether Adaptive Learning is available (either for an instance
+        or for a course).
+
+        If the course_id parameter is passed, it returns a boolean indicating
+        whether Adaptive Learning is available for the particular course.
+        Note that in order for Adaptive Learning to be available in a given course
+        both flags (an instance-wide and and a course-specific ones)
+        need to exist and be enabled.
```

Comment:
@tomaszgy There's still some typos here: The part in parentheses should read "(an instance-wide flag and a course-specific one)".

Reply:
👍